### DISCUSSION TITLE
How to troubleshoot vm not starting

### DISCUSSION THREAD

**Discussion by kuba-moo** [arighi/virtme-ng#45] [General]
I have a custom kernel build, I used to put it into a VM to do my testing, I wanted to cut out the installation steps and give virtme-ng a go. But even tho (I think) I enabled all the config options the prompt never appears. The command just hangs, qemu takes 100% of CPU and I have to kill the process.

Is there an easy way to surface the kernel logs while the vm boots?

**Comment by arighi** [comment]
Have you tried to use `vng -v` (verbose mode)? With `-v` you should see all the boot messages, maybe that can help to figure out what's going on.

You can also try to generate a minimal .config for virtme-ng using `vng -b --kconfig`, recompile the kernel and see if it boots. If that's the case maybe the reason of the failure is that you're missing a required config option.

**Reply by kuba-moo** [reply]
I did a `vng -b` and that works fine. There's some output as things get built, and in the end it plonks me into a shell with the new kernel.

When I try `vng -v -r build_mm/vmlinux` (where the vmlinux is my custom build) - there's no output, just fans spinning up on my machine :(

**Reply by arighi** [reply]
@kuba-moo you can't run vmlinux directly, you can either use `vng -r arch/x86/boot/bzImage` (assuming you're on x86) or simply use `vng` and it'll figure out the proper boot file automatically (assuming you're in the linux source dir).

**Reply by kuba-moo** [reply]
bzImage works, thanks a lot!!
